src/Approximations/overapproximate.jl

```diff
@@ -1635,3 +1635,66 @@ function overapproximate(X::LazySet{N}, ::Type{<:Zonotope},
     end
     return Zonotope(c, G)
 end
+
+"""
+    overapproximate(r::Rectification{N, <:AbstractZonotope{N}}, ::Type{<:Zonotope}) where {N}
+
+Overapproximation of the rectification of a zonotopic set.
+
+### Input
+
+- `r` -- lazy rectification of a zonotopic set
+- `Zonotope` -- type for dispatch
+
+### Output
+
+A zonotope overapproximation of the set obtained by rectifying `Z`.
+
+### Algorithm
+
+This function implements [Theorem 3.1, 1].
+
+[1] *Singh, G., Gehr, T., Mirman, M., Püschel, M., & Vechev, M. (2018). Fast
+and effective robustness certification. In Advances in Neural Information
+Processing Systems (pp. 10802-10813).*
+"""
+function overapproximate(r::Rectification{N, <:AbstractZonotope{N}}, ::Type{<:Zonotope}) where {N}
+    Z = copy(r.X)
+    c = center(Z)
+    G = genmat(Z)
+    n, m = size(G)
+    H = overapproximate(Z, Hyperrectangle)
+    row_idx = Vector{Int}()
+    μ_idx = Vector{N}()
+
+    @inbounds for i in 1:n
+        lx, ux = H.center[i] - H.radius[i], H.center[i] + H.radius[i]
+        if !_leq(lx, zero(N))
+            nothing
```

Comment:
`G[i, :]` stays undefined here.

Reply:
What do you mean? G is defined in line 1664

Reply:
Oh, I see.